src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11685,10 +11695,13 @@ private ExpressionSyntax ParsePostFixExpression(ExpressionSyntax expr)
         private bool CanStartConsequenceExpression()
         {
             Debug.Assert(this.CurrentToken.Kind == SyntaxKind.QuestionToken);
-            var nextTokenKind = this.PeekToken(1).Kind;
+            var nextToken = this.PeekToken(1);
+            var nextTokenKind = nextToken.Kind;
 
-            // ?. is always the start of of a consequence expression.
-            if (nextTokenKind == SyntaxKind.DotToken)
+            // ?.   is always the start of of a consequence expression.
+            //
+            // ?..  is a ternary with a range expression as it's 'whenTrue' clause.
```

Comment:
Consider adding a positive test for this scenario (I only saw an error scenario with `?..`)

Reply:
Positive tests added in RangeExpressionParsingTests.